### PR TITLE
Add  dcp validation

### PR DESCRIPTION
**Pull request by olbjo**
Implementation of validation functions and test cases

### CODE REVIEW

File: src/device/pf_cmina.c [hunk 1]
```diff
@@ -829,3 +853,150 @@ void pf_cmina_show(
...
+ * - Station-names do not have the form a.b.c.d with a, b, c, d = 0...999
+ *
+ * Also empty station name is valid here (indicating that no station name has been set).
+ *
+ * @param station_name     In: Station name
+ * @param len              In: Length of incoming string without terminator.
+ * @return  0  if the name is valid
+ *          -1 if the name is invalid
+ */
+int pf_cmina_is_stationname_valid(
```
Comment (hefloryd) [inline src/device/pf_cmina.c:885]: These validation functions effectively compute boolean values and should probably return bool?

Comment (olbjo) [reply src/device/pf_cmina.c:885]: OK, changed return type to bool

File: src/device/pf_cmina.c [hunk 2]
```diff
@@ -880,12 +881,100 @@ void pf_cmina_show(
...
    uint16_t                len)
 {
-   /* Todo: Implement */
+   uint16_t i;
+   uint16_t label_offset = 0;
+   char prev = 'a';
+   uint16_t n_labels = 1;
+   uint16_t n_non_digit = 0;
+
+   if ((!station_name) && (len > 0))
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_cmina.c:895]: OK

File: src/device/pf_cmina.c [hunk 3]
```diff
@@ -880,12 +881,100 @@ void pf_cmina_show(
...
+   {
+      return 0;
+   }
+
+   if ((station_name[0] == '-') || (len > 240))
+   {      
+      return -1;
+   }
+
+   if (0 == strncmp(station_name, "port-", 5))
```
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_cmina.c:910]: OK

File: src/device/pf_cmina.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (olbjo) [reply src/device/pf_cmina.c]: Now updated with comments!

**Pull request merged**